src/Lachain.Core/Blockchain/VM/ExecutionFrame/FrameFactory.cs

```diff
@@ -31,7 +32,8 @@ public class FrameFactory
             ulong gasLimit)
         {
             frame = new WasmExecutionFrame(
-                WasmExecutionFrame.CompileWasm(currentAddress, code, blockchainInterface.GetFunctionImports()),
+                WasmExecutionFrame.CompileWasm(currentAddress, code, blockchainInterface.GetFunctionImports(), 
+                    context.Snapshot.Blocks.GetTotalBlockHeight() > Hardfork.HardforkHeights.Hardfork_2),
```

Comment:
We should fix on some convention. 
For staking Contract I used the following convention: 
0...(Hardfork1 - 1) -> first algorithm
Hardfork1 .... everything -> second algorithm 

https://github.com/LATOKEN/lachain/blob/0e518513325accb2e08f60cbb250d4b4dcaa0b3f/src/Lachain.Core/Blockchain/SystemContracts/StakingContract.cs#L678

We should follow the same convention for every hardforks.